soda-core/src/soda_core/cli/handlers/data_source.py

```diff
@@ -47,7 +47,9 @@ def handle_test_data_source(data_source_file_path: str) -> ExitCode:
     soda_logger.info(f"Testing data source configuration file {data_source_file_path}")
     from soda_core.common.data_source_impl import DataSourceImpl
 
-    data_source_impl: DataSourceImpl = DataSourceImpl.from_yaml_source(YamlSource.from_file_path(data_source_file_path))
+    data_source_impl: DataSourceImpl = DataSourceImpl.from_yaml_source(
+        DataSourceYamlSource.from_file_path(data_source_file_path)
```

Comment:
the only point of DataSourceYamlSource subclass is to have the type constant set by default so that we don't manipulate the state of the object somewhere between parsing and creating data structures